responsibleai/responsibleai/databalanceanalysis/distribution_balance_measures.py

```diff
@@ -0,0 +1,274 @@
+    """
+    Returns the cross entropy between two probability distributions.
+    https://en.wikipedia.org/wiki/Cross_entropy
+
+    :param obs: The observed probability distribution.
+    :type obs: np.array
+    :param ref: The reference probability distribution.
+    :type ref: np.array
+    :return: The cross entropy between the two probability distributions.
+    :rtype: float
+    """
+    return -np.sum(ref * np.log2(obs))
+
+
+def get_kl_divergence(obs: np.array, ref: np.array) -> float:
+    """
+    Returns the Kullback-Leibler (KL) divergence, a measure of how one
+    probability distribution is different from another.
+    https://en.wikipedia.org/wiki/Kullback%E2%80%93Leibler_divergence
+
+    :param obs: The observed probability distribution.
+    :type obs: np.array
+    :param ref: The reference probability distribution.
+    :type ref: np.array
+    :return: The KL divergence between the two probability distributions.
+    :rtype: float
+    """
+    return scipy.stats.entropy(obs, qk=ref)
+
+
+def get_js_distance(obs: np.array, ref: np.array) -> float:
+    """
+    Returns the Jenson-Shannon Distance, a measure that is the
+    symmetrized and smoothed version of the KL divergence.
+    https://en.wikipedia.org/wiki/Jensen%E2%80%93Shannon_divergence
+
+    :param obs: The observed probability distribution.
+    :type obs: np.array
+    :param ref: The reference probability distribution.
+    :type ref: np.array
+    :return: The JS Distance between the two probability distributions.
+    :rtype: float
+    """
+    avg = (obs + ref) / 2
+    divergence = (
+        scipy.stats.entropy(obs, avg) + scipy.stats.entropy(ref, avg)
+    ) / 2
+    distance = np.sqrt(divergence)
+    return distance
+
+
+def get_ws_distance(obs: np.array, ref: np.array) -> float:
+    """
+    Returns the Wasserstein Distance, a measure that symbolizes the
+    minimum cost to turn one probability distribution into another.
+    https://en.wikipedia.org/wiki/Wasserstein_metric
+
+    :param obs: The observed probability distribution.
+    :type obs: np.array
+    :param ref: The reference probability distribution.
+    :type ref: np.array
+    :return: The Wasserstein Distance between the two probability
+        distributions.
+    :rtype: float
+    """
+    return scipy.stats.wasserstein_distance(obs, ref)
+
+
+def get_inf_norm_dist(obs: np.array, ref: np.array) -> float:
+    """
+    Returns the Infinity norm distance (AKA the Chebyshev distance),
+    a measure of the maximum absolute difference between two probability
+    distributions.
+    https://en.wikipedia.org/wiki/Chebyshev_distance
+
+    :param obs: The observed probability distribution.
+    :type obs: np.array
+    :param ref: The reference probability distribution.
+    :type ref: np.array
+    :return: The Infinity norm distance between the two probability
+        distributions.
+    :rtype: float
+    """
+    return scipy.spatial.distance.chebyshev(obs, ref)
```

Comment:
I completely see that point; I think Ilya also pointed out in a comment that these measures can potentially be moved to `fairlearn`. 

I actually hadn't thought about data balance measures having a home somewhere like `scipy` mainly because our team was focused on advertising these as part of the Responsible AI effort.

Thinking more about it, the measures by themselves don't tell a coherent story. In trying to create an end-to-end story behind why these measures matter, I think visualizing it is the first step, and giving actionable insights is the second step.

For example, a dashboard giving a tooltip such as "Hey, we noticed that your dataset sees a lot more positive labels when Gender == Male compared to when Gender == Female, so your model may be more likely to correlate Gender == Male with the positive label. Here's a small code snippet you can run to better balance these two feature values: `...`".

When we do have a coherent story of measures + insights, I'd love to pick your brain about where everything belongs. 

Then again, it probably makes sense to figure out the home for these measures before the insights part is built so that they can be consumed properly. @martthalch do you mind capturing this for our future backlog so we can make sure it's discussed? 😄